### PR TITLE
Fix invalid unreachable code error following a panic statement in unreachable block

### LINKED ISSUE #36916: Invalid "unreachable code" error
**Description:**
$title.

**Steps to reproduce:**
```ballerina
function fn(float? v) returns int {
    int y = 0;

    if v is () {
        return 1;
    } else if v is float {
        
    } else {
        panic error("err!");
    }

    return y; // says unreachable code, but reachable.
}
```

**Affected Versions:**
2201.1.0

### PR DESCRIPTION
**Pull request by lasinicl**
## Purpose
$title

Fixes #36916

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by MaryamZi** [comment]
```ballerina
function fn(float? v) returns int {
    if v is () {
        return 1;
    } else if v is float {
        return 2;
    } else {
        string message = "err!";
        panic error(message);
    }
}
```

seems to result in an extra `this function must return a result` error now. We could maybe fix this separately since there's unreachable code anyway.

File: tests/jballerina-unit-test/src/test/resources/test-src/reachability-analysis/reachability_analysis.bal [hunk 1]
```diff
@@ -1176,6 +1176,73 @@ function testReachabilityWithQueryAction() {
...
+
+    if v is () {
+        panic error("Error");
+    } else if v is float {
+
+    } else {
+        panic error("err!");
+    }
+
+    return y;
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/reachability-analysis/reachability_analysis.bal:1230]: I'm not sure if this works atm, but we should ideally have tests like

```ballerina
    int y;

    if v is () {
        y = 1;
    } else if v is float {
        y = 2;
    } else {
        panic error("err!");
    }

    return y;
```

**Review by MaryamZi — APPROVED** [review]

**Comment by MaryamZi** [comment]
We need to send a PR to master with these changes too.

File: tests/jballerina-unit-test/src/test/resources/test-src/reachability-analysis/reachability_analysis.bal (hunk 1 above)
Comment (lasinicl) [reply tests/jballerina-unit-test/src/test/resources/test-src/reachability-analysis/reachability_analysis.bal:1230]: This works as expected with this fix. Will add tests and address the above issue via a separate PR.

**Review by gimantha — APPROVED** [review]

**Comment by lasinicl** [comment]
PR to master - https://github.com/ballerina-platform/ballerina-lang/pull/36940

**Pull request merged**